addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -944,3 +974,36 @@ def mark_as_todo(self):
         self._compute_state()
 
         return
+
+
+def get_next_name(obj, name):
+    """
+    Get the next name for an object.
+
+    For when we want to create an object whose name links back to a previous
+    object.  For example BATCH/00001-02.
+    Assumes original names are of the form `r".*\d{5}"`.
+
+    Arguments:
+        obj - the source object for the name
+        name - the name for the object's model in the ir_sequence table
+
+    Returns:
+        The generated name, a string.
+    """
+    IrSequence = obj.env['ir.sequence']
+
+    # Name pattern for continuation object.
+    # Is two digits enough?
+    ir_sequence = IrSequence.search([('name', '=', name)], limit=1)
+    name_pattern = r'({}\d{{5}})-(\d{{2}})'.format(re.escape(ir_sequence.prefix))
```

Comment:
Can we change '\d{{5}}' to any number of digits? Because when the sequence reaches 99999 will carry on to six digits anyway. Also 5 is default lenght, but not always since it is a setting.